tests/unit/modules/test_win_network.py

```diff
@@ -312,3 +311,20 @@ def test_get_route(self):
                     "source": "10.0.0.15",
                 },
             )
+
+    def test_connect_53371(self):
+        """
+        Test that UnboundLocalError is not thrown on socket.gaierror
+        as reported in #53371
+        """
+        with patch(
+            "socket.getaddrinfo",
+            MagicMock(side_effect=socket.gaierror("[Errno 11004] getaddrinfo failed")),
```

Comment:
**suggestion**: pretty sure we want autospec here. Also can just provide `side_effect` as an argument to patch rather than creating a MagicMock.
```suggestion
            autospec=True,
            side_effect=socket.gaierror("[Errno 11004] getaddrinfo failed"),
```

Otherwise, 👍

Reply:
Thanks, I've applied your suggestion and checked if pre-commit is fine with that. I've also updated the branch to latest master, so test could run with all current content.